atc/db/pipeline_lifecycle.go

```diff
@@ -40,20 +40,31 @@ func (pl *pipelineLifecycle) ArchiveAbandonedPipelines() error {
 		Where(sq.And{
 			// pipeline was set by some build
 			sq.NotEq{"p.parent_job_id": nil},
+			// pipeline is not already archived
+			sq.Eq{"p.archived": false},
 			sq.Or{
 				// job (that set child pipeline) from parent pipeline is
 				// removed, Concourse marks job as inactive
 				sq.Eq{"j.active": false},
 				// parent pipeline was destroyed, entire job record is gone
 				sq.Eq{"j.id": nil},
-				// parent pipeline was archived
+				// // parent pipeline was archived
 				sq.Eq{"parent.archived": true},
-				// build that set the pipeline is not the most recent for the job.
-				// parent_build_id can be later than latest_completed_build_id if this
-				// gc query runs during a run of a build, specifically between the time
-				// of a completed set pipeline step and the build finishing
-				sq.Expr("p.parent_build_id < j.latest_completed_build_id"),
-			}}).
+				// // build that set the pipeline is not the most recent for the job.
+				// // parent_build_id can be later than latest_completed_build_id if this
+				// // gc query runs during a run of a build, specifically between the time
+				// // of a completed set pipeline step and the build finishing
```

Comment:
should revert the changed lines?

Reply:
Hmm I don't think the changed lines here introduced the bug 🤔  I feel like the bug was just caused by the thing this pr is fixing but I could be wrong and missing something?

Reply:
I mean the double commented lines. @clarafu

Reply:
Ohhhh haha yes good catch 😆